### PR TITLE
Quick morning refactor

### PR DESCRIPTION
**Pull request by timnallen**
Hey guys,

Looking through, made a couple of changes. We can stash them if you don't like them. I DRYed up the cart nav bar, added a test for admin editing user profile data (the way we were testing, it didn't test that the form was re-rendered if the info was missing, and it only re-rendered if the email was taken, so I fixed that in the controller).

I also replaced our hard-coded flash messages with the active record generated ones, which saves several lines of code. I'm fine with not doing it this way, but I thought I'd see what you think.

Check it out, and let me know.

### CODE REVIEW

File: spec/features/admins/admin_user_sees_user_profile_spec.rb [hunk 1]
```diff
@@ -53,12 +53,33 @@
...
 
       expect(page).to have_content("User profile has been updated")
-      expect(page).to have_content("Username: editedname")
+      expect(page).to have_content("Name: editedname")
       expect(page).to have_content("Email: new_email@email.com")
 
       expect(current_path).to eq(admin_user_path(user))
     end
 
+    it 'wont allow me to update without all the right info' do
```
Comment (teresa-m-knowles) [inline spec/features/admins/admin_user_sees_user_profile_spec.rb:62]: Thanks for testing this sad path! didn't think of it last night, makes total sense

**Review by teresa-m-knowles — APPROVED** [review]
I like this, thanks for the work @timnallen

**Comment by csvlewis** [comment]
I like it! The dynamic error messages are nice and they should do a lot of the work for us.

**Review by csvlewis — APPROVED** [review]

**Review by allieIsOkay — APPROVED** [review]
Looks good.

**Pull request merged**